File: files/Factorial.cpp
#include <stdio.h>
#include <iostream.h>
#include <conio.h>


//using namespace std;
FILE *in = fopen("C:/entrada.txt","r");
FILE *out = fopen("C:/salidaCpp.txt","w");
int main(){    
    int caseN;
	fscanf(in,"%d",&caseN);              //lee una linea como entero lo guarda en la variable caseN
	for(int i=0;i<caseN;i++){
       int numero;
            	fscanf(in,"%d",&numero);
            	int fact=1;
            	for(int j=1;j<numero;j++){
                        fact=fact*j;
                        }
                fprintf(out,"Caso #%d: %d\n",i+1,fact);
            }
}

File: files/Factorial.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace rcsharp
{
    class Factorial
    {
        static void Main(string[] args)        {
                using (StreamReader sr = new StreamReader("C:/entrada.txt"))
                using (StreamWriter sw = new StreamWriter("C:/salidaCsharp.out"))
                {
                    int n = int.Parse(sr.ReadLine());
                    for (int i = 0; i < n; i++)
                    {
                        int numero = int.Parse(sr.ReadLine());
                        int fact = 1;
                        for (int j = 1; j < numero; j++)
                        {
                            fact = fact * j;
                        }
                        sw.WriteLine("Caso #" + (i + 1) + ": " + fact);
                    }
                }
        }
    }
}
